### PR TITLE
Implemented edit credentials functionality

### PR DESCRIPTION
**Pull request by ajaypratap003**
### Type of change

- Enhancement / new feature
- Refactoring

### Description
1. Create serialize  function for credentials
2. Create deserialize  function for credentials
3. Integrate mock-server to get exising credentials and polputae fields in edit cerdentials form
4. Save credentials

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 1]
```diff
@@ -236,7 +235,7 @@ const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
         <DeviceDetailLayout />
         <SplitItem isFilled />
         <SplitItem className={css(styles.kebab_toggle_margin)}>
-          <Switch
+          <SwitchWithToggle
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:238]: I had replaced `SwitchWithToggle` with patternfly's `Switch` for an issue that is coming up again.
device_detail
Even if we cancel in the dialog, the status gets changed.

File: console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 2]
```diff
@@ -114,16 +111,18 @@ const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
           </SplitItem>
         </Split>
         <Flex className={css(styles.namespace_info_margin)}>
+          <FlexItem>
```
Comment (obabec) [inline console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:114]: Add ID for this flex item.

File: console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 3]
```diff
@@ -236,7 +235,7 @@ const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
         <DeviceDetailLayout />
         <SplitItem isFilled />
         <SplitItem className={css(styles.kebab_toggle_margin)}>
-          <Switch
+          <SwitchWithToggle
             id="switch-device-header-enable-btn"
```
Comment (obabec) [inline console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:239]: Try to unify ids so it in most of the console is used 'button' change this ID to  switch-device-header-enable-button

File: console/console-init/ui/src/modules/iot-device/containers/EditCredentialsContainer/edit-credentials.module.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.marginLeft {
+  margin-left: 25px;
```
Comment (obabec) [inline console/console-init/ui/src/modules/iot-device/containers/EditCredentialsContainer/edit-credentials.module.css:2]: I am not professional in UI design but this concrete margins never work :D What if user will have 4k ultrawide monitor? I would go with some more flexible units like rem or % for example.

File: console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 5]
```diff
@@ -13,13 +13,14 @@ import {
...
   PageSection,
-  PageSectionVariants
+  PageSectionVariants,
+  Divider,
+  Switch
 } from "@patternfly/react-core";
 import { StyleSheet, css } from "aphrodite";
 import { FormatDistance } from "use-patternfly";
-import { DropdownWithKebabToggle } from "components";
+import { DropdownWithKebabToggle, SwitchWithToggle } from "components";
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:23]: Remove this unused variable `SwitchWithToggle`.

**Pull request merged**